### ISSUE TITLE
msbuild incomplete / broken

### ISSUE THREAD

**Issue by fluffynuts** [gentoo/dotnet#415]
relates to:
- https://github.com/gentoo/dotnet/issues/379
- https://github.com/gentoo/dotnet/issues/405

issues experienced:
- msbuild attempts to build to / (so the user will see output as per issue #379)
- many targets missing (so, eg, MVC applications will fail to build)
- cannot build with Jetbrains Rider (xbuild has some other issue and msbuild, which is now recommended, doesn't work)

msbuild in Ubuntu seems to be in a correct state though. I'm not sure how to fix this, but the Ubuntu package might be a good place to look for differences.

Please let me know if there's some way I can help with this.

**Comment by cnd** [comment]
cc: @ArsenShnurkov

**Comment by ArsenShnurkov** [comment]
Which version of ebuild you have tried? What repository you compiled? What was the text of the error?

That's correct, that some/many targets are missing, you just need to package them.

**Comment by ShadowCat8** [comment]
@ArsenShnurkov 
I had seen a couple questions that you had asked earlier on this thread, and, I can say that I was trying to use dev-util/msbuild-15.3-r3 from the "dotnet" overlay on Gentoo.

I have the output from enabling the "+debug" flag on my build of msbuild to [the thread](https://forums.gentoo.org/viewtopic-p-8258212.html#8258212) in the Gentoo forums.

Hope that helps.  If you need anything else that I can provide, let me know.

**Comment by ArsenShnurkov** [comment]
> If you need anything else that I can provide, let me know.

please try https://github.com/ArsenShnurkov/shnurise/blob/master/dev-util/msbuild/msbuild-15.9.20.62856-r1.ebuild
but that will not solve your problem, because someone need to package Sdk's

**Comment by ostroffjh** [comment]
@ArsenShnurkov I have tried that ebuild, and get an error that src/MSBuild/mono-MSBuild.csproj: '>' is an unexpected token. The expected token is '='. Line 44, position 87.   I think I fixed that by changing the end of that line from > to />
However, then, (even with debug and developer flags set) I get error for missing .cs files under ../src/Build/:
  Debugger/DebuggerLocalType.cs
  Debugger/DebuggerManager.cs
  BackEnd/Components/SdkResolution/SdkResolution.cs
  Collections/OrdinalKeyedComparer.cs
  Collections/OrdinalIgnoreCaseKeyedComparer.cs
I have no idea if that is part of the Sdks packaging issue, or something else.

**Comment by ArsenShnurkov** [comment]
no, it's something else. I will recheck the ebuild itself (because mine compiles)

**Comment by ostroffjh** [comment]
[build.log](https://github.com/gentoo/dotnet/files/2824949/build.log)
Build log attached.  I think those missing .cs files were bogus, because I still get the earlier error, even with the patch to change the > to />

**Comment by ostroffjh** [comment]
OK, I'm now feeling pretty stupid. I finally figured out I was sloppy downloading the two .csproj files, and had ended up with html, not the real csproj.eix msbuild.  Fixed those, and the compile finished fine.  The install failed because of a sandbox issue (I'll raise that later.)  The final qmerge still fails due to a file conflict, but I suspect I need to get the eselect working first, so I have more to do before posting again.

**Comment by ArsenShnurkov** [comment]
> I need to get the eselect working first

I wrote <a href="https://github.com/ArsenShnurkov/shnurise/blob/b4ffa8022e67a6fd2be9b0c986467d99ef3a5075/app-eselect/eselect-msbuild/eselect-msbuild-15.9.ebuild">eselect-msbuild</a>, but msbuild ebuild is not ready yet (i need to finish msbuild's configuration, to help it find default tasks)

**Comment by Ristovski** [comment]
@ArsenShnurkov Is there any progress on this?

**Comment by ArsenShnurkov** [comment]
@Ristovski - too many dependencies to repackage (for example https://github.com/gentoo/dotnet/issues/422), I need help from other gentoo developers

**Comment by filmor** [comment]
msbuild should probably be removed or hard-masked in its current form. The ebuild is essentially non-functional and severely outdated as well.
